apps/shipments/views.py

```diff
@@ -206,3 +206,18 @@ def create(self, request, *args, **kwargs):
         permission_link = PermissionLink.objects.create(**serializer.validated_data)
 
         return Response(PermissionLinkSerializer(permission_link).data, status=status.HTTP_201_CREATED)
+
+
+class ShipmentHistoryListView(viewsets.ViewSet):
+    permission_classes = ((IsOwnerOrShared,) if settings.PROFILES_ENABLED
+                          else (permissions.AllowAny,))
+    http_method_names = ['get', ]
+
+    def list(self, request, *args, **kwargs):
```

Comment:
Due to the complexity of ChangesDiffSerializer, we will almost certainly want to cache the results of this view. We might even should pre-cache the view when a shipment is updated.